microcosm_daemon/daemon.py

```diff
@@ -125,7 +125,11 @@ def start(self):
         """
         self.initialize()
         self.graph.logger.info("Starting daemon {}".format(self.name))
-        self.run_state_machine()
+
+        try:
+            self.run_state_machine()
+        finally:
+            exit(0)
```

Comment:
What if the daemon actually terminated in error? Won't this make it look like everything is fine?

Reply:
Hey!!

That's a good point! I have assumed, It should be terminated earlier, with `exit(non-zero)`. But to be fair, I should rather do a lovely try/else, so it's clear from the perspective of the daemon loop:

```
try:
    self.run_state_machine()
except:
    # log err
    exit(1)
else:
    exit(0)
```

Reply:
Yeah that looks good to me 👍😊